src/components/Data/Documents/Page.vue

```diff
@@ -26,16 +28,49 @@
             }"
             >Create New Document</b-button
           >
+          <b-dropdown
+            class="ml-1"
+            data-cy="Refresh-dropdown"
+            split
+            variant="outline-primary"
+            :title="
+              autoSync
+                ? 'Documents are updated in real-time'
+                : 'Refresh the list to apply pending changes'
+            "
+            @click="fetchDocuments"
+          >
+            <template #button-content>
+              <i
+                class="fas fa-sync mr-1"
+                data-cy="Autosync-icon"
+                :class="{ 'fa-spin': autoSync, 'text-secondary': !autoSync }"
+              ></i>
+              Refresh
+            </template>
+            <b-dropdown-item
+              data-cy="Autosync-toggle"
+              :disabled="!displayRealtimeButton.includes(listViewType)"
+              @click="autoSync = !autoSync"
+            >
+              <i
+                :class="
+                  `far ${autoSync ? 'fa-check-square' : 'fa-square'} left`
+                "
+              />
+              Toggle auto-sync
```

Comment:
A check box is already a way to toggle a feature
```suggestion
              Auto-sync
```